torch/csrc/jit/passes/onnx.cpp

```diff
@@ -226,7 +226,94 @@ void BlockToONNX(
 
   // Finally, visit all nodes in the graph
   for (auto node : old_block->nodes()) {
-    if (node->kind() == prim::PythonOp) {
+    if (node->kind().is_caffe2()) {
+      auto new_node = ctx.block->appendNode(
+          ctx.block->owningGraph()->createClone(node, envFn));
+      for (size_t i = 0; i < node->outputs().size(); i++) {
+        env[node->outputs()[i]] = new_node->outputs()[i];
+      }
+      const auto& schema = node->schema();
+      size_t tensor_input_num = 0;
+      size_t input_index = 0;
+      const auto& args = schema.arguments();
+      for (size_t arg_index = 0; arg_index < args.size(); ++arg_index, ++input_index) {
+        auto type = args[arg_index].type();
+        if (type->kind() == TypeKind::OptionalType) {
+          const auto& input_value = node->inputs()[input_index];
+          if (input_value->mustBeNone()) {
+            new_node->removeInput(tensor_input_num);
+            continue;
+          } else {
+            type = reinterpret_cast<OptionalType*>(type.get())->getElementType();
+            // recursive optional type is not supported
+            AT_ASSERT(type->kind() != TypeKind::OptionalType);
+          }
+        }
+        if (type->isSubclass(TypeKind::TensorType)) {
+          AT_ASSERT(input_index == tensor_input_num);
+          ++tensor_input_num;
+          continue;
+        } else if (type->kind() == TypeKind::BoolType || type->kind() == TypeKind::FloatType || type->kind() == TypeKind::IntType) {
+          const auto* constant_node = node->inputs()[input_index]->node();
+          AT_ASSERT(constant_node->kind() == prim::Constant);
```

Comment:
Those shouldn't be AT_ASSERT (which indicate system failure) but rather AT_CHECK (user-input failure) and please include more descriptive message - e.g. "Inputs to scalar values Caffe2 operator wrappers should be literals"